### PR TITLE
[ch240] Add additional category to data entry form for user testing

### PR DESCRIPTION
**Pull request by seerocode**
## Story details
https://app.clubhouse.io/stanford-computational-policy-lab/story/240

### Changes:
- add disability category to test data 
- group entries by category sections in data entry form (using lodash)
- add tests for sections 

### Screenshots:
image

image

### CODE REVIEW

**Comment by seerocode** [comment]
Description for disability above was taken from the CDC website: https://www.cdc.gov/ncbddd/disabilityandhealth/disability.html

File: api/database.py [hunk 1]
```diff
@@ -303,39 +303,64 @@ def create_dummy_data(session):
     tag.datasets.append(ds1)
     tag.datasets.append(ds2)
 
-    category_non_binary = Category(id='51349e29-290e-4398-a401-5bf7d04af75e', description='Non-binary gender',
+    # Gender category
+    gender_description = 'Gender identity expresses one\'s innermost concept of self as male,\
```
Comment (jnu) [inline api/database.py]: not a problem, but fyi python has a couple of other options for multiline strings:

option 1: triple quotes - use the slash to escape the line breaks:
```py
gender_description = """\
Gender identity expresses one's innermost concept of self as male, \
female, a blend of both or neither - how individuals perceive \
...
"""
```

option 2: magical string concatenation -- no need to escape anything, all the lines are implicitly concatenated as if you used "+"
```py
gender_description = (
  "Gender identity expresses one's innermost concept of self as male, "
  "female, a blend of both or neither - how individuals perceive "
  ...
)
```

File: client/src/components/DataEntry/DataEntryAggregateDataEntryForm.css [hunk 2]
```diff
@@ -1,17 +1,17 @@
 .data-entry-form_input-grid {
   display: grid;
-  grid-template-columns: 1fr 1fr;
+  grid-template-columns: repeat(auto-fill, minmax(min(16rem, 100%), 1fr));
```
Comment (jnu) [inline client/src/components/DataEntry/DataEntryAggregateDataEntryForm.css:3]: 😰 

css grids still blow my mind -- curious what browsers you've tested this on? i think it's pretty widely supported these days, but just making sure

File: client/src/components/DataEntry/DataEntryCategorySections.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+interface DataEntryCategorySection {
+  entries: Entry[];
+  onValueChange: (event: ChangeEvent<HTMLInputElement>, index: number) => void;
+}
+
+const groupByCategory = (entries: Entry[]) => {
+  const categoryGroups = _(entries)
+    // Group the elements of Array based on nested `category` property
+    .groupBy((obj) => obj.category)
```
Comment (jnu) [inline client/src/components/DataEntry/DataEntryCategorySections.tsx:15]: <3 this is so clean, thanks for adding lodash! i hope it felt faster to write as well

File: client/src/components/DataEntry/DataEntryCategorySections.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  return categoryGroups;
+};
+
+const DataEntryCategorySections = ({
+  entries,
+  onValueChange,
+}: DataEntryCategorySection): JSX.Element => {
+  const { t } = useTranslation();
+
+  const handleChange = (
```
Comment (jnu) [inline client/src/components/DataEntry/DataEntryCategorySections.tsx]: curious what this function is doing? looks like you could just pass in onValueChange directly, no?

**Review by jnu — APPROVED** [review]
looks great, nice work!

File: api/database.py (hunk 1 above)
Comment (seerocode) [reply api/database.py]: Is there a significant difference between triple quote and single quote or is it more for meaning like "this is a long string here"?

File: client/src/components/DataEntry/DataEntryAggregateDataEntryForm.css (hunk 2 above)
Comment (seerocode) [reply client/src/components/DataEntry/DataEntryAggregateDataEntryForm.css:3]: Works great in Safari, Chrome, and Firefox on my end! Can I Use gives more detail: https://caniuse.com/css-grid

IE is still a bummer.

File: client/src/components/DataEntry/DataEntryCategorySections.tsx (hunk 3 above)
Comment (seerocode) [reply client/src/components/DataEntry/DataEntryCategorySections.tsx:15]: Yes and more clear/readable for sure!

File: client/src/components/DataEntry/DataEntryCategorySections.tsx (hunk 4 above)
Comment (seerocode) [reply client/src/components/DataEntry/DataEntryCategorySections.tsx]: You're right! Just removed it.

File: api/database.py (hunk 1 above)
Comment (seerocode) [reply api/database.py]: NVM, just checked myself. I changed it to triple single quotes. Your comment led me to this article which was also helpful: https://www.digitalocean.com/community/tutorials/how-to-format-text-in-python-3

**Pull request merged**